### DISCUSSION TITLE
For loop not working

### DISCUSSION THREAD

**Discussion by memxor** [aptos-labs/aptos-developer-discussions#42] [Questions] [answered]
![image](https://github.com/aptos-labs/aptos-developer-discussions/assets/48633453/98d20213-e2f3-405f-936a-c3bf4800f15f)

I'm having issues with for loops, I've the latest version of Aptos CLI, but for some reason this is not working. Any ideas?

**Comment by gregnazario** [comment, marked as answer]
Hey, this looks like a bug in the compiler, but there is a simple solution.

Simply change this broken code here
```move
for(iter in 0..length) {
  // ...
}
```

And add a space between `for` and `(`.  `for(` -> `for (`
```move
for (iter in 0..length) {
  // ...
}
```

There is a bug reported here https://github.com/aptos-labs/aptos-core/issues/11921
